src/components/Button.js

```diff
@@ -41,15 +43,13 @@ module.exports = React.createClass({
 		};
 	},
 	render() {
+		var classes = this.props.sheet.classes;
```

Comment:
1. I also like decorator syntax for this more, but technically its exactly the same. I would be happy to start using babel with experimental features for this project. Also it doesn't forces to use all es6 features everywhere at once.
1. Not sure what you mean. What is exactly confusing about getting props? Seems to be normal react flow. In fact you can manage it within component if you want to. react-jss is an optional tool. Its just you need to write some lines of code in every component.
1. If someone can prove that we don't need to attach/detach styles when they are not in use and that a very big amount of styles doesn't slow down the application - we can get rid of react-jss. This helper project is based upon an assumption that selector engine and node insertion gets slower when there are lots of css selectors to consider. I am still searching for a way how to bench this. https://github.com/jsstyles/jss/issues/48